internal/apm/php.go

```diff
@@ -103,13 +107,17 @@ func (i *PhpInjector) Inject(ctx context.Context, inst current.Instrumentation,
 	// caller checks if there is at least one container.
 	container := &pod.Spec.Containers[firstContainer]
 
+	// set a blank value, so that php will scan the config dir that was configured during compilation with --with-config-file-scan-dir
+	// do this first so we can override the behavior later.  We only set it if it was already blank or the key was not defined
+	if val, ok := getValueFromEnv(container.Env, envIniScanDirKey); !ok || val == "" {
+		setEnvVar(container, envIniScanDirKey, "", true)
+	}
 	setEnvVar(container, envIniScanDirKey, envIniScanDirVal, true)
 
 	// inject PHP instrumentation spec env vars.
 	for _, env := range inst.Spec.Agent.Env {
-		if idx := getIndexOfEnv(container.Env, env.Name); idx == -1 {
-			container.Env = append(container.Env, env)
-		}
+		_, shouldConcat := appendPathsEnvKeys[env.Name]
+		setEnvVar(container, env.Name, env.Value, shouldConcat)
```

Comment:
This will concatenate the values again and is not needed since it was already handled here: https://github.com/newrelic/k8s-agents-operator/pull/262/files#diff-2595818f8907aefd0e1f264a562deb12d0dedf8ef3d741a80ff86b08ff47fc1bR115

Reply:
I see your point that setEnvVar does the same sort of append for non-existing values, but it's not necessary here.
append worked fine for all values since  PHP_INI_SCAN_DIR is the only one that needed special handling and PHP_INI_SCAN_DIR is not something that can be (or should be) set  in the instrumentation crd.  PHP_INI_SCAN_DIR is a PHP defined env var that customers (if they set) should do it where they are setting up their PHP application not when they are trying to get agent instrumentation.